sdk.adoc

```diff
@@ -0,0 +1,111 @@
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+----
+
+== Overview
+The purpose of uProtocol language specific SDKs are to provide the consistent mental model across all programming languages (Java, C++, Rust, Python, etc...) for core uP-L1 & L2 specifications, interfaces, and data models. The SDK is core to all uProtocol software as it is used to implements uEs (apps & services), transports, and platforms 
+
+The following specification shall cover the requirements that all implementations of the uProtocol SDK *MUST* adhere to.
+
+NOTE: SDKs *MUST* be released under a permissive license (Apache 2.0, MIT, etc..) and located in https://github.com/eclipse-uprotocol[GitHub Eclipse-uProtocol] project.
+
+
+== Source 
+The following section will elaborate on the source code requirements of the SDK that includes the content (modules) and the folder and namespace structures.
+
+* The root namespace (package name) *MUST* be `org.eclipse.uprotocol`
+* Modules defined in <<content>> below *MUST* be implemented in their own folder (namespace)
+* Each module *MUST* divide the implementation of the specification as defined in <<folders>> below
+
+* The uProtocol data model *MUST* be declared in protos and each SDK imports the artifacts from the proto library
+
+
+.Module Folders/Namespace
+[#folders,width="80%",cols="20%,80%",options="header"]
+|===
+| Folder | Description
+
+| `validators`
+| static methods to validate the data model
+
+| `serializer`
+| Any code that is responsible to Serialize and deserialize the data model
+
+| `factory` (or `builder`)
+| Factory methods to build the data model per business logic (i.e. to build CloudEvents, UUIDs, etc..)
+
+|===
+
+
+=== Modules
+
+Modules are packages (folders) that reside under the sdk's `src` director. <<content>> table below lists the required modules.
+
+.SDK Modules
+[#content,width="100%",cols="20%,20%,60%",options="header"]
+|===
+| Folder | Specification |Description
+
+| `uri`
+| link:basics/uri.adoc[UUri]
+| How things are addressed in uProtocol
+
+| `uuid`
+| link:basics/uuid.adoc[UUID]
+| Unique ID & timestamp specification 
+
+| `cloudevents`
+| link:up-l1/cloudevents.adoc[CloudEvents]
+| Specification of how uProtocol data models can be expressed using CloudEvents data model
+
+| `transport`
+| link:up-l1/README.adoc[uTransport]
+| uP-L1 uTransport interface & data model definitions used to build language specific transports implementations (i.e. zenoh, SOME/IP, MQTT, etc..) that run on a given uPlatform.
+
+| `rpcclient`
```

Comment:
Aren't these uTransport, rpcClient and RPCServer and interfaces which need to be implemented in the uLink?
The SDK libraries are at the end, the specific implementation.

Reply:
Rephased and addressed